### DISCUSSION TITLE
render `.ftd` without any `ftd section` as markdown

### DISCUSSION THREAD

**Discussion by AbrarNitk** [fastn-stack/fastn#735] [Ideas & RFCs]
Feature: [markup](https://github.com/FifthTry/fpm/discussions/449)

<details><summary>Status: Planned</summary>

[Discord Thread](https://discord.com/channels/793929082483769345/1011900045454483486/1011901349627183104)

</details>

If I write in ftd document like below, browser screen shows blank.

```ftd
I am text without using any ftd component.
```

If I use ftd component like so it renders in browser

```ftd 
-- ds.markdown:

I am text with using ftd component doc-site.markdown.
```

So `ftd` document should default `markdown`.

**Comment by Heulitig** [comment]
If we do this then 
```markdown
<some text without section here>   <- This will be converted to markdown 

-- section: 
some header 
...

some section body

<some text without section here>     <- This will be included in the above section body 

--- sub-section:
some headers 

some subsection body 2
```

So based on your suggestion this default markdown text will only be usable at the top of any `ftd` document only (prior to the start of any section). So is this idea feasible or are we going to implement this in some other way.

**Reply by amitu** [reply]
This is only for `.ftd` **without** `section`s. Your example contains sections so it will go through regular ftd processing. Only documents that zero sections defined would be treated as markdown files.

**Reply by amitu** [reply]
To handle missing top level section we will have to do more hacky stuff that we do not want to. With no-section check is a cheap check that fpm can perform and ftd would not need any changes for this.
